modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/viewmodel/ProfileUpgradeBannerViewModel.kt

```diff
@@ -105,4 +133,50 @@ class ProfileUpgradeBannerViewModel @Inject constructor(
             settings.setLastSelectedSubscriptionTier(upgradeFeatureCard.subscriptionTier)
         }
     }
+
+    private fun getDefaultSubscription(
+        filteredSubscriptions: List<Subscription>,
+        cachedTier: SubscriptionTier?,
+        cachedSubscriptionStatus: SubscriptionStatus?,
+    ) = subscriptionManager.getDefaultSubscription(
+        subscriptions = filteredSubscriptions,
+        tier = if (cachedTier == SubscriptionTier.PATRON) {
+            Subscription.SubscriptionTier.PATRON
+        } else {
+            Subscription.SubscriptionTier.PLUS
+        },
+        frequency = getSubscriptionFrequency(cachedSubscriptionStatus)
+    )
+
+    private fun getSubscriptionFrequency(cachedSubscriptionStatus: SubscriptionStatus?) =
+        if (cachedSubscriptionStatus is SubscriptionStatus.Paid) {
+            when (cachedSubscriptionStatus.frequency) {
+                SubscriptionFrequency.NONE -> SubscriptionFrequency.YEARLY
+                SubscriptionFrequency.MONTHLY -> SubscriptionFrequency.MONTHLY
+                SubscriptionFrequency.YEARLY -> SubscriptionFrequency.YEARLY
+            }
+        } else {
+            SubscriptionFrequency.YEARLY
+        }
```

Comment:
Same as above, I like being more explicit with a `when` that covers all the cases explicitly.

``` kotlin
when (cachedSubscriptionStatus) {
    is SubscriptionStatus.Paid -> {
        when (cachedSubscriptionStatus.frequency) {
            SubscriptionFrequency.NONE -> SubscriptionFrequency.YEARLY
            SubscriptionFrequency.MONTHLY -> SubscriptionFrequency.MONTHLY
            SubscriptionFrequency.YEARLY -> SubscriptionFrequency.YEARLY
        }
    }

    null,
    is SubscriptionStatus.Free -> SubscriptionFrequency.YEARLY
}

```

Same as above, not a blocker. 🙂

Reply:
Same as above, updated it in https://github.com/Automattic/pocket-casts-android/commit/a6138221c0f11b64e2b4b2825ac032ffa4cd0b2f